src/main/kotlin/io/quarkus/code/GitHubResource.kt

```diff
@@ -0,0 +1,47 @@
+package io.quarkus.code
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.quarkus.code.model.QuarkusProject
+import io.quarkus.code.services.CodeQuarkusConfigManager
+import io.quarkus.code.services.GitHubService
+import io.quarkus.code.services.QuarkusProjectCreator
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import org.eclipse.microprofile.openapi.annotations.Operation
+import java.io.IOException
+import javax.inject.Inject
+import javax.validation.Valid
+import javax.ws.rs.*
+import javax.ws.rs.core.MediaType.APPLICATION_JSON
+import javax.ws.rs.core.Response
+
+@Path("/github")
+class GitHubResource {
+
+    @Inject
+    lateinit var gitHubService: GitHubService
+
+    @Inject
+    lateinit var projectCreator: QuarkusProjectCreator
+
+    @GET
+    @Path("/push")
+    @Produces(APPLICATION_JSON)
+    @Operation(summary = "Push generated code to GitHub")
+    fun pushCode(@Valid @BeanParam project: QuarkusProject, @HeaderParam("token") token: String): Response {
+        val location = projectCreator.createTmp(project)
+        val repo = gitHubService.createRepository(token, project.artifactId)
+        gitHubService.push(repo.first, token, repo.second, location)
+        return Response.ok("{ \"repository\": \"${repo.second}\"}").build()
+    }
+
+    @GET
+    @Path("/auth")
+    @Produces(APPLICATION_JSON)
+    fun authenticate(@QueryParam("code") code: String, @QueryParam("state") state: String): Response {
+        val token = gitHubService.fetchAccessToken(code, state)
+        return Response.ok("{ \"token\": \"$token\"}").build()
```

Comment:
If you return a bean containing a token, it would encode it to json. I guess you could justify doing it your way for perf..